semantic-core/src/Core/Core.hs

```diff
@@ -233,7 +232,7 @@ annAt :: (Carrier sig m, Member (Ann ann) sig) => ann -> m a -> m a
 annAt ann = send . Ann ann
 
 annWith :: (Carrier sig m, Member (Ann Span) sig) => CallStack -> m a -> m a
-annWith callStack = maybe id (annAt . snd) (stackLoc callStack)
+annWith callStack = maybe id (annAt . spanFromSrcLoc . snd) (listToMaybe (getCallStack callStack))
```

Comment:
```suggestion
annWith callStack = fromMaybe (annAt . spanFromSrcLoc . snd) (listToMaybe (getCallStack callStack))
```

Reply:
`fromMaybe x` = `maybe x id`, not `maybe id x`, so this won’t work.